query/src/main/java/io/keen/client/java/KeenQueryClient.java

```diff
@@ -0,0 +1,704 @@
+        QueryResult result = execute(queryParams, timeframe);
+        return queryResultToDouble(result);
+    }
+
+    /**
+     * Average query with only the required arguments.
+     * Query API info here: https://keen.io/docs/api/reference/#average-resource
+     *
+     * @param eventCollection     The name of the event collection you are analyzing.
+     * @param targetProperty     The name of the property you are analyzing.
+     * @param timeframe     The {@link RelativeTimeframe} or {@link AbsoluteTimeframe}.
+     * @return The average query response.
+     * @throws IOException If there was an error communicating with the server or
+     * an error message received from the server.
+     */
+    public Double average(String eventCollection, String targetProperty, Timeframe timeframe) throws IOException  {
+        Query queryParams = new Query.QueryBuilder(QueryType.AVERAGE_RESOURCE)
+                .withEventCollection(eventCollection)
+                .withTargetProperty(targetProperty)
+                .build();
+        QueryResult result = execute(queryParams, timeframe);
+        return queryResultToDouble(result);
+    }
+
+    /**
+     * Median query with only the required arguments.
+     * Query API info here: https://keen.io/docs/api/reference/#median-resource
+     *
+     * @param eventCollection     The name of the event collection you are analyzing.
+     * @param targetProperty     The name of the property you are analyzing.
+     * @param timeframe     The {@link RelativeTimeframe} or {@link AbsoluteTimeframe}.
+     * @return The median query response.
+     * @throws IOException If there was an error communicating with the server or
+     * an error message received from the server.
+     */
+    public Double median(String eventCollection, String targetProperty, Timeframe timeframe) throws IOException  {
+        Query queryParams = new Query.QueryBuilder(QueryType.MEDIAN_RESOURCE)
+                .withEventCollection(eventCollection)
+                .withTargetProperty(targetProperty)
+                .build();
+        QueryResult result =  execute(queryParams, timeframe);
+        return queryResultToDouble(result);
+    }
+
+    /**
+     * Percentile query with only the required arguments.
+     * Query API info here: https://keen.io/docs/api/reference/#percentile-resource
+     *
+     * @param eventCollection     The name of the event collection you are analyzing.
+     * @param targetProperty     The name of the property you are analyzing.
+     * @param percentile     The percentile.
+     * @param timeframe     The {@link RelativeTimeframe} or {@link AbsoluteTimeframe}.
+     * @return The percentile query response.
+     * @throws IOException If there was an error communicating with the server or
+     * an error message received from the server.
+     */
+    public Double percentile(String eventCollection, String targetProperty, Double percentile, Timeframe timeframe) throws IOException  {
```

Comment:
Nit: Is there an reason for the `percentile` parameter to be a `Double` instead of a `double` here? The former allows it to be `null` but I would think we don't want to allow that.